OpenId Connect Dev Console fails when setting quarkus.http.root-path

### Describe the bug

We want to set `quarkus.http.root-path` since we're putting our application behind a reverse proxy.
We also want to use the OpenId Connect Dev Console.

Unfortunately it seems the latter breaks when we set the former.
The resulting error is:

```
2025-07-04 10:23:11,857 ERROR [io.ver.ext.web.RoutingContext] (vert.x-eventloop-thread-4) Unhandled exception in router: io.quarkus.security.AuthenticationFailedException
        at io.quarkus.oidc.runtime.DefaultTokenStateManager.decryptToken(DefaultTokenStateManager.java:247)
        at io.quarkus.oidc.runtime.DefaultTokenStateManager.getTokens(DefaultTokenStateManager.java:119)
        at io.quarkus.oidc.runtime.DefaultTokenStateManager_ClientProxy.getTokens(Unknown Source)
        at io.quarkus.oidc.runtime.dev.ui.OidcDevSessionCookieReaderHandler.handle(OidcDevSessionCookieReaderHandler.java:26)
        at io.quarkus.oidc.runtime.dev.ui.OidcDevSessionCookieReaderHandler.handle(OidcDevSessionCookieReaderHandler.java:14)
        at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1283)
        at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
        at io.vertx.ext.web.impl.RoutingContextWrapper.next(RoutingContextWrapper.java:205)
        at io.quarkus.vertx.http.runtime.filters.accesslog.AccessLogHandler.handle(AccessLogHandler.java:147)
        at io.quarkus.vertx.http.runtime.filters.accesslog.AccessLogHandler.handle(AccessLogHandler.java:94)
        at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1283)
        at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
        at io.vertx.ext.web.impl.RoutingContextWrapper.next(RoutingContextWrapper.java:205)
        at io.vertx.ext.web.impl.RouterImpl.handleContext(RouterImpl.java:251)
        at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1283)
        at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
        at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:151)
        at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup.handleHotReplacementRequest(VertxHttpHotReplacementSetup.java:146)
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder$HotReplacementRoutingContextHandler.handle(VertxHttpRecorder.java:1714)
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder$HotReplacementRoutingContextHandler.handle(VertxHttpRecorder.java:1704)
        at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1283)
        at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
        at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:151)
        at io.vertx.ext.web.impl.RoutingContextImpl.route(RoutingContextImpl.java:100)
        at io.vertx.ext.web.impl.RouterImpl.handle(RouterImpl.java:69)
        at io.vertx.ext.web.impl.RouterImpl.handle(RouterImpl.java:37)
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder$7.handle(VertxHttpRecorder.java:537)
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder$7.handle(VertxHttpRecorder.java:534)
        at io.quarkus.vertx.http.runtime.options.HttpServerCommonHandlers$2.handle(HttpServerCommonHandlers.java:87)
        at io.quarkus.vertx.http.runtime.options.HttpServerCommonHandlers$2.handle(HttpServerCommonHandlers.java:70)
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder$1.handle(VertxHttpRecorder.java:187)
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder$1.handle(VertxHttpRecorder.java:161)
        at io.vertx.core.impl.ContextImpl.emit(ContextImpl.java:342)
        at io.vertx.core.impl.DuplicatedContext.emit(DuplicatedContext.java:163)
        at io.vertx.core.http.impl.Http1xServerConnection.handleMessage(Http1xServerConnection.java:174)
        at io.vertx.core.net.impl.ConnectionBase.read(ConnectionBase.java:159)
        at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:153)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
        at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:289)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1357)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:868)
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:796)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:732)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:658)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.NullPointerException: Cannot invoke "io.quarkus.oidc.runtime.TenantConfigContext.getSessionCookieEncryptionKey()" because "configContext" is null
        at io.quarkus.oidc.runtime.DefaultTokenStateManager.decryptToken(DefaultTokenStateManager.java:245)
        ... 61 more
```

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/oidc/deployment/src/main/java/io/quarkus/oidc/deployment/devservices/AbstractDevUIProcessor.java
+++ b/extensions/oidc/deployment/src/main/java/io/quarkus/oidc/deployment/devservices/AbstractDevUIProcessor.java
@@ -76,6 +76,10 @@ public abstract class AbstractDevUIProcessor {
             graphqlUiPath = null;
         }
 
+        final String devUiLogoutPath = nonApplicationRootPathBuildItem.resolvePath("io.quarkus.quarkus-oidc/logout");
+        final String devUiReadSessionCookiePath = nonApplicationRootPathBuildItem
+                .resolvePath("io.quarkus.quarkus-oidc/readSessionCookie");
+
         cardPage.addBuildTimeData("devRoot", nonApplicationRootPathBuildItem.getNonApplicationRootPath());
 
         RuntimeValue<OidcDevUiRpcSvcPropertiesBean> runtimeProperties = recorder.getRpcServiceProperties(
@@ -83,7 +87,7 @@ public abstract class AbstractDevUIProcessor {
                 keycloakUsers, oidcProviderName, oidcApplicationType, oidcGrantType,
                 introspectionIsAvailable, keycloakAdminUrl, keycloakRealms, swaggerIsAvailable,
                 graphqlIsAvailable, swaggerUiPath, graphqlUiPath, alwaysLogoutUserInDevUiOnReload, discoverMetadata,
-                authServerUrl);
+                authServerUrl, devUiLogoutPath, devUiReadSessionCookiePath);
 
         recorder.createJsonRPCService(beanContainer.getValue(), runtimeProperties);
 

--- a/extensions/oidc/deployment/src/main/resources/dev-ui/qwc-oidc-provider.js
+++ b/extensions/oidc/deployment/src/main/resources/dev-ui/qwc-oidc-provider.js
@@ -49,7 +49,9 @@ class OidcPropertiesState extends LitState {
             propertiesStateId: null,
             testServiceResponses: null,
             webAppLoginObserver: null,
-            isWebApp: false
+            isWebApp: false,
+            logoutPath: null,
+            readSessionCookiePath: null
         };
     }
 
@@ -88,6 +90,8 @@ class OidcPropertiesState extends LitState {
             propertiesState.keycloakRealms = response.result.keycloakRealms;
             propertiesState.swaggerUiPath = response.result.swaggerUiPath;
             propertiesState.graphqlUiPath = response.result.graphqlUiPath;
+            propertiesState.logoutPath = response.result.logoutPath;
+            propertiesState.readSessionCookiePath = response.result.readSessionCookiePath;
 
             return {
                 // logout === true will trigger query params removal
@@ -257,7 +261,6 @@ export class QwcOidcProvider extends QwcHotReloadElement {
     constructor() {
         super();
         this._devRoot = (devRoot?.replaceAll('/', '%2F') ?? '') + 'dev-ui'; // e.g. /q/dev-ui
-
         this._selectedRealm = null;
         this._servicePath = '/';
         this._selectedClientId = null;
@@ -390,7 +393,7 @@ export class QwcOidcProvider extends QwcHotReloadElement {
     }
 
     _signInToService() {
-        window.open("http://localhost:" + propertiesState.httpPort + this._servicePath);
+        window.open(QwcOidcProvider._toServiceUrl(this._servicePath));
     }
 
     static _isServiceOrHybridApp() {
@@ -914,8 +917,7 @@ export class QwcOidcProvider extends QwcHotReloadElement {
     _getEncodedPath() {
         // this is the last part of this path: /q/dev-ui/io.quarkus.quarkus-oidc/keycloak-provider -> keycloak-provider
         const subPath = window.location.pathname.substring(window.location.pathname.lastIndexOf('/') + 1);
-        return "http%3A%2F%2Flocalhost%3A" + propertiesState.httpPort + this._devRoot
-            + "%2Fio.quarkus.quarkus-oidc%2F" + subPath;
+        return QwcOidcProvider._getEncodedCurrentBaseUrl() + this._devRoot + "%2Fio.quarkus.quarkus-oidc%2F" + subPath;
     }
 
     _getClientId() {
@@ -1091,7 +1093,7 @@ export class QwcOidcProvider extends QwcHotReloadElement {
                 tokenUrl = tokenUrl.substring(0, index + 8) + parts[1] + "/protocol/openid-connect/token";
                 clientId = parts[2];
             }
-            const redirectUri = "http://localhost:" + propertiesState.httpPort + window.location.pathname;
+            const redirectUri = QwcOidcProvider._getCurrentBaseUrl() + window.location.pathname;
             const clientSecret = propertiesState.clientSecret;
             jsonRpc
                 .exchangeCodeForTokens({tokenUrl, clientId, clientSecret, authorizationCode, redirectUri})
@@ -1175,10 +1177,16 @@ export class QwcOidcProvider extends QwcHotReloadElement {
         });
     }
 
+    static _getEncodedCurrentBaseUrl() {
+        return QwcOidcProvider._getCurrentBaseUrl().replaceAll('/', '%2F').replaceAll(':', '%3A')
+    }
+
+    static _getCurrentBaseUrl() {
+        return window.location.origin
+    }
+
     static _checkSessionCookie(jsonRpc, onLoggedIn, onLoggedOut) {
-        // FIXME: hardcoded path?
-        const port = propertiesState.httpPort ?? 8080
-        fetch("http://localhost:" + port + "/q/io.quarkus.quarkus-oidc/readSessionCookie")
+        fetch(QwcOidcProvider._getCurrentBaseUrl() + propertiesState.readSessionCookiePath)
            .then(response => response.json())
                 .then(result => {
                     if ("id_token" in result || "access_token" in result) {
@@ -1194,7 +1202,7 @@ export class QwcOidcProvider extends QwcHotReloadElement {
                         } else {
                             propertiesState.idToken = null;
                         }
-                        propertiesState.logoutUrl = "http://localhost:8080/q/io.quarkus.quarkus-oidc/logout";
+                        propertiesState.logoutUrl = QwcOidcProvider._getCurrentBaseUrl() + propertiesState.logoutPath;
                         propertiesState.postLogoutUriParam = "redirect_uri";
                         onLoggedIn();
                     } else {

--- a/extensions/oidc/runtime-dev/src/main/java/io/quarkus/oidc/runtime/dev/ui/OidcDevJsonRpcService.java
+++ b/extensions/oidc/runtime-dev/src/main/java/io/quarkus/oidc/runtime/dev/ui/OidcDevJsonRpcService.java
@@ -30,7 +30,8 @@ public class OidcDevJsonRpcService {
                 props.isIntrospectionIsAvailable(), props.getKeycloakAdminUrl(),
                 props.getKeycloakRealms(), props.isSwaggerIsAvailable(), props.isGraphqlIsAvailable(), props.getSwaggerUiPath(),
                 props.getGraphqlUiPath(),
-                props.isAlwaysLogoutUserInDevUiOnReload(), props.getPropertiesStateId());
+                props.isAlwaysLogoutUserInDevUiOnReload(), props.getPropertiesStateId(),
+                props.getLogoutPath(), props.getReadSessionCookiePath());
     }
 
     public Uni<String> exchangeCodeForTokens(String tokenUrl, String clientId, String clientSecret,

--- a/extensions/oidc/runtime-dev/src/main/java/io/quarkus/oidc/runtime/dev/ui/OidcDevUiRecorder.java
+++ b/extensions/oidc/runtime-dev/src/main/java/io/quarkus/oidc/runtime/dev/ui/OidcDevUiRecorder.java
@@ -41,7 +41,7 @@ public class OidcDevUiRecorder {
             Map<String, String> oidcUsers, String oidcProviderName, String oidcApplicationType, String oidcGrantType,
             boolean introspectionIsAvailable, String keycloakAdminUrl, List<String> keycloakRealms, boolean swaggerIsAvailable,
             boolean graphqlIsAvailable, String swaggerUiPath, String graphqlUiPath, boolean alwaysLogoutUserInDevUiOnReload,
-            boolean discoverMetadata, String authServerUrl) {
+            boolean discoverMetadata, String authServerUrl, String devUiLogoutPath, String devUiReadSessionCookiePath) {
         if (discoverMetadata) {
             JsonObject metadata = discoverMetadata(authServerUrl);
             if (metadata != null) {
@@ -56,7 +56,8 @@ public class OidcDevUiRecorder {
                 new OidcDevUiRpcSvcPropertiesBean(authorizationUrl, tokenUrl, logoutUrl,
                         webClientTimeout, grantOptions, oidcUsers, oidcProviderName, oidcApplicationType, oidcGrantType,
                         introspectionIsAvailable, keycloakAdminUrl, keycloakRealms, swaggerIsAvailable,
-                        graphqlIsAvailable, swaggerUiPath, graphqlUiPath, alwaysLogoutUserInDevUiOnReload));
+                        graphqlIsAvailable, swaggerUiPath, graphqlUiPath, alwaysLogoutUserInDevUiOnReload,
+                        devUiLogoutPath, devUiReadSessionCookiePath));
     }
 
     public Handler<RoutingContext> readSessionCookieHandler() {

--- a/extensions/oidc/runtime-dev/src/main/java/io/quarkus/oidc/runtime/dev/ui/OidcDevUiRpcSvcPropertiesBean.java
+++ b/extensions/oidc/runtime-dev/src/main/java/io/quarkus/oidc/runtime/dev/ui/OidcDevUiRpcSvcPropertiesBean.java
@@ -29,13 +29,16 @@ public class OidcDevUiRpcSvcPropertiesBean {
      * and web component needs to be updated.
      */
     private final String propertiesStateId;
+    private final String logoutPath;
+    private final String readSessionCookiePath;
 
     OidcDevUiRpcSvcPropertiesBean(String authorizationUrl, String tokenUrl, String logoutUrl,
             Duration webClientTimeout, Map<String, Map<String, String>> grantOptions,
             Map<String, String> oidcUsers, String oidcProviderName, String oidcApplicationType,
             String oidcGrantType, boolean introspectionIsAvailable, String keycloakAdminUrl,
             List<String> keycloakRealms, boolean swaggerIsAvailable, boolean graphqlIsAvailable,
-            String swaggerUiPath, String graphqlUiPath, boolean alwaysLogoutUserInDevUiOnReload) {
+            String swaggerUiPath, String graphqlUiPath, boolean alwaysLogoutUserInDevUiOnReload,
+            String devUiLogoutPath, String devUiReadSessionCookiePath) {
         this.authorizationUrl = authorizationUrl;
         this.tokenUrl = tokenUrl;
         this.logoutUrl = logoutUrl;
@@ -54,6 +57,8 @@ public class OidcDevUiRpcSvcPropertiesBean {
         this.graphqlUiPath = graphqlUiPath;
         this.alwaysLogoutUserInDevUiOnReload = alwaysLogoutUserInDevUiOnReload;
         this.propertiesStateId = Long.toString(UUID.randomUUID().getLeastSignificantBits());
+        this.logoutPath = devUiLogoutPath;
+        this.readSessionCookiePath = devUiReadSessionCookiePath;
     }
 
     String getAuthorizationUrl() {
@@ -134,5 +139,13 @@ public class OidcDevUiRpcSvcPropertiesBean {
 
     String getPropertiesStateId() {
         return propertiesStateId;
+    }
+
+    String getLogoutPath() {
+        return logoutPath;
+    }
+
+    String getReadSessionCookiePath() {
+        return readSessionCookiePath;
     }
 }

--- a/extensions/oidc/runtime-dev/src/main/java/io/quarkus/oidc/runtime/dev/ui/OidcDevUiRuntimePropertiesDTO.java
+++ b/extensions/oidc/runtime-dev/src/main/java/io/quarkus/oidc/runtime/dev/ui/OidcDevUiRuntimePropertiesDTO.java
@@ -37,12 +37,15 @@ public class OidcDevUiRuntimePropertiesDTO {
     private final String graphqlUiPath;
     private final boolean alwaysLogoutUserInDevUiOnReload;
     private final String propertiesStateId;
+    private final String logoutPath;
+    private final String readSessionCookiePath;
 
     OidcDevUiRuntimePropertiesDTO(String authorizationUrl, String tokenUrl, String logoutUrl, Config config, int httpPort,
             String oidcProviderName, String oidcApplicationType, String oidcGrantType,
             boolean introspectionIsAvailable, String keycloakAdminUrl, List<String> keycloakRealms,
             boolean swaggerIsAvailable, boolean graphqlIsAvailable, String swaggerUiPath,
-            String graphqlUiPath, boolean alwaysLogoutUserInDevUiOnReload, String propertiesStateId) {
+            String graphqlUiPath, boolean alwaysLogoutUserInDevUiOnReload, String propertiesStateId,
+            String logoutPath, String readSessionCookiePath) {
         this.clientId = new OidcConfigPropertySupplier(CLIENT_ID_CONFIG_KEY).get(config);
         this.clientSecret = new OidcConfigPropertySupplier(CLIENT_SECRET_CONFIG_KEY, "").get(config);
         this.authorizationUrl = new OidcConfigPropertySupplier(AUTHORIZATION_PATH_CONFIG_KEY, authorizationUrl, true)
@@ -65,6 +68,8 @@ public class OidcDevUiRuntimePropertiesDTO {
         this.graphqlUiPath = graphqlUiPath;
         this.alwaysLogoutUserInDevUiOnReload = alwaysLogoutUserInDevUiOnReload;
         this.propertiesStateId = propertiesStateId;
+        this.logoutPath = logoutPath;
+        this.readSessionCookiePath = readSessionCookiePath;
     }
 
     public String getClientId() {
@@ -149,5 +154,13 @@ public class OidcDevUiRuntimePropertiesDTO {
 
     public String getPropertiesStateId() {
         return propertiesStateId;
+    }
+
+    public String getLogoutPath() {
+        return logoutPath;
+    }
+
+    public String getReadSessionCookiePath() {
+        return readSessionCookiePath;
     }
 }

